ui/src/common/actions.ts

```diff
@@ -143,6 +160,88 @@ function removeTrack(state: StateDraft, trackId: string) {
   state.pinnedTracks = state.pinnedTracks.filter((id) => id !== trackId);
 }
 
+// A helper to clean the state for a given removable track group.
+function removeTrackGroup(state: StateDraft, groupId: string) {
+  delete state.trackGroups[groupId];
+
```

Comment:
nit: unnecessary empty line.

Reply:
In general I find there are too few blank lines in this code for legibility. But I'll take this one under advisement.

Reply:
Yeah, this is a two-line function. No need for the blank.